### PR TITLE
Port yuzu-emu/yuzu#3398: "Use config mode for finding SDL2 with CMake"

### PR DESCRIPTION
**Pull request by FearlessTobi**
See yuzu-emu/yuzu#3398 for more details.

**Original description:**
FindSDL2.cmake fails for the most recent version of `libsdl2-dev` on Debian. Since SDL2 has included [a bundled CMake package](https://hg.libsdl.org/SDL/file/e723c663ba58/sdl2-config.cmake.in) since 2016, using FindSDL2.cmake should no longer be necessary. See [here](https://stackoverflow.com/a/54006461/9157187) for more information.
Fixes yuzu-emu/yuzu#3397.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5095)

### CODE REVIEW

**Comment by FearlessTobi** [comment]
CI failure is unrelated.

File: CMakeLists.txt [hunk 1]
```diff
@@ -155,15 +155,16 @@ if (ENABLE_SDL2)
...
-    endif()
 
-    if (SDL2_FOUND)
-        # TODO(yuriks): Make FindSDL2.cmake export an IMPORTED library instead
         add_library(SDL2 INTERFACE)
         target_link_libraries(SDL2 INTERFACE "${SDL2_LIBRARY}")
         target_include_directories(SDL2 INTERFACE "${SDL2_INCLUDE_DIR}")
+    else()
+        find_package(SDL2 REQUIRED)
+        include_directories(${SDL2_INCLUDE_DIRS})
```
Comment (BreadFish64) [inline CMakeLists.txt]: should this also be `target_include_directories`?

Comment (B3n30) [reply CMakeLists.txt]: Imo yes

**Comment by Subv** [comment]
The mingw build is failing on Travis:

```
../src/audio_core/sdl2_sink.cpp:7:10: fatal error: SDL.h: No such file or directory
```

**Comment by FearlessTobi** [comment]
I'm not planning on finishing this PR for now.

**Pull request closed without merging**